### PR TITLE
Update and fix storybook

### PR DESCRIPTION
**Pull request by fbosquet-azpn**
The purpose is to upgrade the old storybook example.

The old story book in examples/plain-javascript-example was still in version 5 of storybook and Snyk was reporting several vulnerabilities.
Beside it was not longer possible to start storybook using `npm run storybook`, we were getting an error `start-storybook not found`.
The storybook version was so old that it failed to upgrade using the standard `npx storybook@7 upgrade` comande.

I have then created a new storybook from scratch and copy over all existings stories.

I have then fixup the videx-log types hierarchy for Plot/PlotOptions and Track/TrackOptions using Generic programing.
Before those fixes, all the stories in the storybook did not render properly (all empty pages like the movie in issue #182 )

https://github.com/equinor/videx-wellog/assets/68541104/b2692649-c4a8-4da8-b7ad-bf40b54af4a0

### CODE REVIEW

**Comment by GuillaumeVix** [comment]
Thanks for doing the changes and creating the pull request, we will have a look at it!

**Comment by fbosquet-azpn** [comment]
I have also taken the opportunity to 
- order of the story  (Plot, Track, Controller, Viewer)
- add a StackedTrack (track 6 named "Formation" in the movie for the Log Controller and Log Viewer)
   this highlight a bug with the horizontal display of the StackedTrack that I am planning to work on next
- create a new story with argument "Log Controller With Args", It may be best to demo the capabilities without multiplying the stories.

File: src/plots/interfaces.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export type DefinedFunction = (y?:number, x?:number) => boolean;
  * @example data => data //default
  * @example data => data.dataSeries[1].dataPoints
  */
-export type DataAccessorFunction = (data: any, plotOptions?: Map<string | number, PlotOptions>) => (PlotData | [PlotData, PlotData] | any); // eslint-disable-line no-use-before-define
+export type DataAccessorFunction = (data: any, PlotOptions?: Map<string | number, PlotOptions>) => (PlotData | [PlotData, PlotData] | any); // eslint-disable-line no-use-before-define
```
Comment (w1nklr) [inline src/plots/interfaces.ts]: Why this change ?

**Review by w1nklr — APPROVED** [review]

File: src/plots/interfaces.ts (hunk 1 above)
Comment (fbosquet-azpn) [reply src/plots/interfaces.ts]: no good reason, I have reverted it.

File: examples/storybook/stories/track.stories.js [hunk 2]
```diff
@@ -148,11 +147,10 @@ export const stackedTrack = () => {
   div.style.width = '100px';
   div.style.background = 'lightgrey';
 
-  const scale = scaleLinear().domain([0, 1000]).range([0, 1000]) as Scale;
+  const scale = scaleLinear().domain([0, 1000]).range([0, 1000]);
 
   const track = new StackedTrack('id', {
-    data: ex4,
-    horizontalLabels: true,
```
Comment (t0oF-azpn) [inline examples/storybook/stories/track.stories.js]: Did you remove this on purpose ?

**Review by w1nklr — APPROVED** [review]

**Review by tomktjemsland — APPROVED** [review]
